test/kubetest2-capi/deployer/deployer.go

```diff
@@ -0,0 +1,240 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package deployer implements the kubetest2 capi deployer
+package deployer
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/spf13/pflag"
+	kinddeployer "sigs.k8s.io/kubetest2/kubetest2-kind/deployer"
+	"sigs.k8s.io/kubetest2/pkg/process"
+	"sigs.k8s.io/kubetest2/pkg/types"
+)
+
+// Name is the name of the deployer
+const Name = "capi"
+
+// New implements deployer.New for capi
+func New(opts types.Options) (types.Deployer, *pflag.FlagSet) {
+	kind, flags := kinddeployer.New(opts)
+	d := &deployer{
+		kind:          kind,
+		commonOptions: opts,
+	}
+	bindFlags(d, flags)
+	return d, flags
+}
+
+// assert that New implements types.NewDeployer
+var _ types.NewDeployer = New
+
+type deployer struct {
+	// generic parts
+	commonOptions types.Options
+	kind          types.Deployer
+	// capi specific details
+	provider            string
+	initProviders       []string
+	kubernetesVersion   string
+	controlPlaneCount   string
+	workerCount         string
+	flavor              string
+	useExistingCluster  bool
+	cniManifest         string
+	kubecfgPath         string
+	upTimeout           string
+	workloadClusterName string
+}
+
+func (d *deployer) Kubeconfig() (string, error) {
+	if d.kubecfgPath != "" {
+		return d.kubecfgPath, nil
+	}
+
+	tmpdir, err := ioutil.TempDir("", "kubetest2-capi")
+	if err != nil {
+		return "", err
+	}
+	d.kubecfgPath = path.Join(tmpdir, "kubeconfig.yaml")
+	args := []string{
+		"get", "kubeconfig", d.workloadClusterName,
+	}
+	clusterctl := exec.Command("clusterctl", args...)
+	lines, err := clusterctl.Output()
+	if err != nil {
+		return "", err
+	}
+	if err := ioutil.WriteFile(d.kubecfgPath, lines, 0600); err != nil {
+		return "", err
+	}
+
+	return d.kubecfgPath, nil
+}
+
+// helper used to create & bind a flagset to the deployer
+func bindFlags(d *deployer, flags *pflag.FlagSet) {
+	flags.StringVar(
+		&d.provider, "provider", "", "--provider flag for clusterctl",
```

Comment:
```suggestion
		&d.provider, "provider", "", "--infrastructure flag for clusterctl",
```
What about renaming the flag into `infrastructure-provider`, `cloud-provider` or something more intuitive for people not aware of the Cluster API internals/terminology